### PR TITLE
Prioritize nompi builds

### PR DESCRIPTION
**Pull request by paulromano**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

-------
One issue I noticed with the recently implemented support for MPI variants is that right now the nompi build is not being prioritized. This seems to be the [recommended practice](https://conda-forge.org/docs/maintainer/knowledge_base.html#message-passing-interface-mpi) and is done in other MPI-enabled packages such as h5py and moab. This PR prioritizes the nompi build by increasing its build number relative to the mpi builds.

### CODE REVIEW

**Review by pshriwise — APPROVED** [review]
Thanks @paulromano! 👍🏻 

Gonna let this sit for a day or so for others' comments.

**Review by kkiesling — APPROVED** [review]
that looks good to me. Can this be tested locally to know if it fixes issue #15?

**Comment by paulromano** [comment]
@kkiesling I'm not sure this will 100% fix the issue. I'm _hoping_ that at least this allows a nompi build (the default) to work, but I suspect that the mpi builds might still be affected by #15 due to the moab run_exports.

**Comment by kkiesling** [comment]
I am going to merge this since it has two approvals and is fairly time-sensitive (fingers crossed it fixes the issue..).

**Comment by kkiesling** [comment]
I wonder if we still needed to rerender even though the this doesn't seem to fall into the categories listed for rerender? The [files](https://anaconda.org/conda-forge/dagmc/files) still have yet to be updated even 4+ hrs post merge. The documentation says this should take 30 min to 2 hrs.

**Comment by kkiesling** [comment]
@conda-forge-admin, please rerender

**Comment by kkiesling** [comment]
Maybe that doesn't work on a closed PR

**Comment by kkiesling** [comment]
Alright, we apparently needed to update this build string to trigger the rebuild:

```
-string: {{ mpi_prefix }}_py{{ CONDA_PY }}h{{ PKG_HASH }}
+string: {{ mpi_prefix }}_py{{ CONDA_PY }}h{{ PKG_HASH }}_{{ build }}
```

**Comment by shimwell** [comment]
> that looks good to me. Can this be tested locally to know if it fixes issue #15?

 
Re testing locally, perhaps this is useful
```bash
cd ~/staged-recipes
python build-locally.py
```
variants can also be tested , more details herer
https://conda-forge.org/docs/maintainer/adding_pkgs.html#running-tests-locally-for-staged-recipes

**Comment by kkiesling** [comment]
> > that looks good to me. Can this be tested locally to know if it fixes issue #15?
> 
> Re testing locally, perhaps this is useful
> 
> ```shell
> cd ~/staged-recipes
> python build-locally.py
> ```
> 
> variants can also be tested , more details herer https://conda-forge.org/docs/maintainer/adding_pkgs.html#running-tests-locally-for-staged-recipes

I think that is only for staged recipes.

**Comment by shimwell** [comment]
> > > that looks good to me. Can this be tested locally to know if it fixes issue #15?
> > 
> > 
> > Re testing locally, perhaps this is useful
> > ```shell
> > cd ~/staged-recipes
> > python build-locally.py
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > variants can also be tested , more details herer https://conda-forge.org/docs/maintainer/adding_pkgs.html#running-tests-locally-for-staged-recipes
> 
> I think that is only for staged recipes.

Are these commands any use for testing

```bash
conda install conda-build
cd openmc-feedstock
conda build recipe/ -c conda-forge
```

**Comment by kkiesling** [comment]
that does work to build locally. I am not sure it will replicate/reveal the issue in #15 but we shall see. I think it just does the same thing that CI does for the most part. And CI had passed before this issue arose.

**Comment by kkiesling** [comment]
ope 😵‍💫 Well I tried that locally and again, despite CI passage, it still doesn't work locally 😵‍💫 

```
Solving environment: ...working... done
Traceback (most recent call last):
  File "/home/kkiesling/miniconda3/envs/test_env/bin/conda-build", line 11, in <module>
    sys.exit(main())
  File "/home/kkiesling/miniconda3/envs/test_env/lib/python3.7/site-packages/conda_build/cli/main_build.py", line 488, in main
    execute(sys.argv[1:])
  File "/home/kkiesling/miniconda3/envs/test_env/lib/python3.7/site-packages/conda_build/cli/main_build.py", line 479, in execute
    verify=args.verify, variants=args.variants, cache_dir=args.cache_dir)
  File "/home/kkiesling/miniconda3/envs/test_env/lib/python3.7/site-packages/conda_build/api.py", line 195, in build
    variants=variants
  File "/home/kkiesling/miniconda3/envs/test_env/lib/python3.7/site-packages/conda_build/build.py", line 3093, in build_tree
    notest=notest,
  File "/home/kkiesling/miniconda3/envs/test_env/lib/python3.7/site-packages/conda_build/build.py", line 2128, in build
    create_build_envs(top_level_pkg, notest)
  File "/home/kkiesling/miniconda3/envs/test_env/lib/python3.7/site-packages/conda_build/build.py", line 1996, in create_build_envs
    channel_urls=tuple(m.config.channel_urls))
  File "/home/kkiesling/miniconda3/envs/test_env/lib/python3.7/site-packages/conda_build/environ.py", line 799, in get_install_actions
    actions = install_actions(prefix, index, specs, force=True)
  File "/home/kkiesling/miniconda3/envs/test_env/lib/python3.7/site-packages/conda/common/io.py", line 88, in decorated
    return f(*args, **kwds)
  File "/home/kkiesling/miniconda3/envs/test_env/lib/python3.7/site-packages/conda/plan.py", line 471, in install_actions
    solver = _get_solver_class()(prefix, channels, subdirs, specs_to_add=specs)
  File "/home/kkiesling/miniconda3/envs/test_env/lib/python3.7/site-packages/conda/core/solve.py", line 109, in __init__
    self.specs_to_add = frozenset(MatchSpec.merge(s for s in specs_to_add))
  File "/home/kkiesling/miniconda3/envs/test_env/lib/python3.7/site-packages/conda/models/match_spec.py", line 478, in merge
    reduce(lambda x, y: x._merge(y, union), group) if len(group) > 1 else group[0]
  File "/home/kkiesling/miniconda3/envs/test_env/lib/python3.7/site-packages/conda/models/match_spec.py", line 478, in <lambda>
    reduce(lambda x, y: x._merge(y, union), group) if len(group) > 1 else group[0]
  File "/home/kkiesling/miniconda3/envs/test_env/lib/python3.7/site-packages/conda/models/match_spec.py", line 509, in _merge
    final = this_component.merge(that_component)
  File "/home/kkiesling/miniconda3/envs/test_env/lib/python3.7/site-packages/conda/models/match_spec.py", line 761, in merge
    % (self.raw_value, other.raw_value))
ValueError: Incompatible component merge:
  - 'mpi_openmpi_*'
  - 'mpi_openmpi_tempest*'
```

**Pull request merged**
